sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -605,4 +553,38 @@ private static String format(Method m) {
   private static String formatType(TypeToken<?> t) {
     return (t == null) ? "null" : ReflectHelpers.TYPE_SIMPLE_DESCRIPTION.apply(t.getType());
   }
+
+  static class ErrorReporter {
```

Comment:
Simplified; now neither a parent nor root reporter is necessary. Might add a root reporter when it becomes necessary (gathering multiple errors).